melange-core/c-decl.dylan

```diff
@@ -1134,7 +1134,7 @@ define method mapped-name
 end method mapped-name;
 
 define method type-name (decl :: <object-declaration>) => (result :: <string>);
-  decl.equated | decl.type.dylan-name;
+  decl.equated | decl.type.type-name;
```

Comment:
1. C-FFI still passes.
2. This change isn't needed, but I thought it was nicer to delegate to the member field's method for this instead of depending on its internal implementation directly.